packages/calcite-components/src/components/sheet/sheet.e2e.ts

```diff
@@ -541,4 +541,144 @@ describe("calcite-sheet properties", () => {
+          closable
+          style="width: 25vw"
+          id="popover"
+        >
+          <div style="padding: 0.5rem 1rem 0.75rem">
+            <p style="margin-top: 0">Example Popover.</p>
+          </div>
+        </calcite-popover>
+        <calcite-button appearance="outline" id="popover-button" icon-start="popup">Example Popover</calcite-button>
+        <calcite-tooltip placement="auto" reference-element="tooltip-auto-ref"> Example Tooltip </calcite-tooltip>
+        <calcite-button appearance="outline" id="tooltip-auto-ref">auto</calcite-button>
+      </div>
+    </calcite-modal>
+    <calcite-button onClick="openComponent('example-sheet')"> Open Sheet </calcite-button>
+  `;
+
+  it("closes a stack of open components sequentially in visual order", async () => {
+    const page = await newE2EPage();
+    await page.setContent(componentStack);
+    await skipAnimations(page);
+
+    async function openAndCheckVisibility(page: E2EPage, element: E2EElement) {
+      element.setProperty("open", true);
+      await page.waitForChanges();
+      expect(await element.isVisible()).toBe(true);
+    }
+
+    const sheet = await page.find("calcite-sheet");
+    await openAndCheckVisibility(page, sheet);
+
+    const firstModal = await page.find("#example-modal");
+    await openAndCheckVisibility(page, firstModal);
+
+    const secondModal = await page.find("#another-modal");
+    await openAndCheckVisibility(page, secondModal);
+
+    async function testInputPicker(page: E2EPage, pickerSelector: string, modal: E2EElement) {
+      const inputPicker = await page.find(pickerSelector);
+      inputPicker.click();
+      await page.waitForChanges();
+      expect(await inputPicker.getProperty("open")).toBe(true);
+
+      await page.keyboard.press("Escape");
+      await page.waitForChanges();
+      expect(await inputPicker.getProperty("open")).toBe(false);
+
+      await page.keyboard.press("Escape");
+      await page.waitForChanges();
+      expect(await modal.isVisible()).toBe(false);
+
+      modal.setProperty("open", true);
+      await page.waitForChanges();
+    }
+
+    await testInputPicker(page, "calcite-input-date-picker", secondModal);
+    await testInputPicker(page, "calcite-input-time-picker", secondModal);
+
+    secondModal.setProperty("open", true);
+    await page.waitForChanges();
+
+    const popoverButton = await page.find("#popover-button");
+    popoverButton.click();
+    await page.waitForChanges();
+    const popover = await page.find("calcite-popover");
+    expect(await popover.getProperty("open")).toBe(true);
+
+    await page.keyboard.press("Escape");
+    await page.waitForChanges();
+    expect(await popover.getProperty("open")).toBe(false);
+
+    async function pressEscapeAndCheckVisibility(page: E2EPage, element: E2EElement, expectedVisibility: boolean) {
+      page.keyboard.press("Escape");
+      await page.waitForChanges();
+      expect(await element.isVisible()).toBe(expectedVisibility);
```

Comment:
This should check all the other elements each time esc is pressed to make sure only the expected element is closed.